src/Tasks/Microsoft.NET.Build.Tasks/GenerateToolsSettingsFile.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using System.Xml.Linq;
+using System.IO;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class GenerateToolsSettingsFile : TaskBase
+    {
+        [Required]
+        public string EntryPointRelativePath { get; set; }
+
+        [Required]
+        public string CommandName { get; set; }
+
+        [Required]
+        public string ToolsSettingsFilePath { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            using (StringWriter writer = new StringWriter())
+            {
+                GenerateDocument(EntryPointRelativePath, CommandName).Save(ToolsSettingsFilePath);
+            }
+        }
+
+        internal static XDocument GenerateDocument(string entryPointRelativePath, string commandName)
+        {
+            return new XDocument(
+                new XDeclaration(version: null, encoding: null, standalone: null),
+                new XElement("DotNetCliTool",
+                      new XElement("Commands",
+                          new XElement("Command",
+                          new XAttribute("Name", commandName),
+                          new XAttribute("EntryPoint", entryPointRelativePath),
+                          new XAttribute("Runner", "dotnet")))));
```

Comment:
Should this be a parameter as well? To the task?

Reply:
If we are not currently running it, I'm fine dropping it. Is it just there by habit, because we had it in CLI (old style) tools?

Reply:
"Runner"? In master today, there is only dotnet available as running on consumer side. And I don't think we need user to override it. Either pack won't support at all, or we can derive it from portable vs self-contain

Reply:
Yes, "Runner". So, we are building the first version limited to only portable apps and you would rather make a change later to support other kinds of apps? Feels like it should not be a lot more to make this a parameter. But, your call.

Reply:
Right, let's remove it. I started thinking about documenting it, and YUK. Undocumented but present features would seem a bad idea.

Reply:
@KathleenDollard just to make it clear. We should not add a parameter to Pack to config runner. However we should keep runner in the settings.xml. right?

Reply:
I'm with Livar; it feels like the task should be generic and not hardcode the runner.  If we do need the task to use a different runner, it's one less code change in the future.